pkgs/applications/terminal-emulators/mlterm/default.nix

```
@@ -1,14 +1,99 @@
-{ stdenv, lib, fetchFromGitHub, pkg-config, autoconf, makeDesktopItem, nixosTests
-, libX11, gdk-pixbuf, cairo, libXft, gtk3, vte
-, harfbuzz #substituting glyphs with opentype fonts
-, fribidi, m17n_lib #bidi and encoding
+{ stdenv
+, lib
+, fetchFromGitHub
+, pkg-config
+, autoconf
+, makeDesktopItem
+, nixosTests
+, vte
+, harfbuzz # can be replaced with libotf
+, fribidi
+, m17n_lib
 , libssh2 #build-in ssh
-, fcitx5, fcitx5-gtk, ibus, uim #IME
+, fcitx5
+, fcitx5-gtk
+, ibus
+, uim #IME
 , wrapGAppsHook #color picker in mlconfig
-, Cocoa #Darwin
+, gdk-pixbuf
+, gtk3
+, gtk ? gtk3
+# List of gui libraries to use. According to `./configure --help` ran on
+# release 3.9.3, options are: (xlib|win32|fb|quartz|console|wayland|sdl2|beos)
+, enableGuis ? {
+  xlib = enableX11;
+  fb = stdenv.isLinux;
+  quartz = stdenv.isDarwin;
+  wayland = stdenv.isLinux;
+  sdl2 = true;
+}
+, libxkbcommon
+, wayland # for the "wayland" --with-gui option
+, SDL2 # for the "sdl" --with-gui option
+# List of typing engines, the default list enables compiling all of the
+# available ones, as recorded on release 3.9.3
+, enableTypeEngines ? {
+  # Considered legacy
+  xcore = false;
```

```suggestion
  xcore = false; # legacy
```

It is clearer, less chance to be conflated as "all below are legacy".

---

> It is clearer, less chance to be conflated as "all below are legacy".

You have a point :).